lib/pipeline.js

```diff
@@ -499,6 +487,42 @@ class PipelineModel extends BaseModel {
         return admin;
     }
 
+    /**
+     * This function deletes admins who does not have proper
+     * GitHub permission, and returns an proper admin.
+     * @method getFirstAdmin
+     * @return {Promise}
+     */
+    async getFirstAdmin() {
+        /* eslint-disable no-restricted-syntax */
+        for (const username of Object.keys(this.admins)) {
```

Comment:
The idea that the order of admin related to width of permission is not reasonable.
That is defined too implicitly, so anyone can't notice the admin hierarchy.
Pipeline's admin will get fuzzy in the context of external permission.

It is just an idea, but solusions of this problem are
1. Add new model for scmList, and each scm repository, not pipline, has admins.
2. Admin should have push permission of all repositories of `scmList`, so admin will be removed if any of the permission of `scmList` goes of.

Reply:
AFAIK `admins` is an object of form `{adminName: true}`, and we are not making use of the object to store anything other than `true` so why not just make it an array ? However not a priority.

I agree this can't bring a solution to `scmList` auth concerns. And it seems that we might get insertion order for free http://2ality.com/2015/10/property-traversal-order-es6.html, assuming `admins` object is constructed similarly each time it's instantiated.